docs/user-guide/global-configurations/docker-registries.md

```diff
@@ -107,4 +107,21 @@ Some popular registries which can be used using username and password mechanism:
 ![](../../user-guide/global-configurations/images/Container_Registry_gcr.jpg)
 
 * **Google Artifact Registry (GAR)** : JSON key file authentication method can be used to authenticate with username and password. Please follow [link](https://cloud.google.com/artifact-registry/docs/docker/authentication#json-key) for getting username and password for this registry. Please remove all the white spaces from json key and wrap it in single quote while putting in password field.
-* **Azure Container Registry (ACR)** : Service principal authentication method can be used to authenticate with username and password. Please follow [link](https://docs.microsoft.com/en-us/azure/container-registry/container-registry-auth-service-principal) for getting username and password for this registry.
\ No newline at end of file
+* **Azure Container Registry (ACR)** : Service principal authentication method can be used to authenticate with username and password. Please follow [link](https://docs.microsoft.com/en-us/azure/container-registry/container-registry-auth-service-principal) for getting username and password for this registry.
+
+## Integrating With External Container Registry
+
+If you want to use a private registry for container registry other than ecr, this will be used to push image and then create a secret in same environment to pull the image to deploy. To create secret, go to charts section and search for chart ‘dt-secrets’ and configure the chart. Provide an App Name and select the Project and Environment in which you want to deploy this chart and then configure the values.yaml as shown in example. The given example is for DockerHub but you can configure similarly for any container registry that you want to use.
+
+```yaml
+name: regcred
+type: kubernetes.io/dockerconfigjson
+labels:
+ test: chart
+secrets:
+ data:
+   - key: .dockerconfigjson
+     value: '{"auths":{"https://index.docker.io/v1/":{"username":"<username>","password":"<password>}}}'
+```     
+
+The `name` that you provide in values.yaml ie. `regcred` is name of the secret that will be used as `imagePullSecrets` to pull the image from docker hub to deploy.
```

Comment:
Also specify how imagePullSecrets will be used in the deployment-template or provide a link to the documentation link https://docs.devtron.ai/devtron/user-guide/creating-application/deployment-template/rollout-deployment#imagepullsecrets

Reply:
Changes done